### PR TITLE
DEVPROD-12086 Add queue sorting factor for number of dependent tasks

### PR DESCRIPTION
**Pull request by hadjri**
DEVPROD-12086

### Description
Introduce a new queue prioritization factor that lets us tune up or tune down the influence the # of dependent tasks has on its placement in the queue. This is settable on the admin page globally, as well as overridable for specific distros.

Since it looks like we've been facing issues with a large number of tasks with unfulfilled dependencies at the front of queues causing next_task route latency issues, if we turn this factor up it might be able to alleviate the issue by fulfilling their dependencies faster.
### Testing
Tested locally

### CODE REVIEW

File: model/distro/distro.go [hunk 1]
```diff
@@ -346,6 +347,13 @@ func (s *PlannerSettings) GetGenerateTaskFactor() int64 {
 	return s.GenerateTaskFactor
 }
 
+func (s *PlannerSettings) GetNumDependentsFactor() int64 {
+	if s.NumDependentsFactor <= 0 {
+		return 10
```
Comment (hadjri) [inline model/distro/distro.go:352]: 10 rather than 1 so we can tune both up and down from what our current behavior is.

File: scheduler/planner.go [hunk 2]
```diff
@@ -258,7 +258,7 @@ func (u *unitInfo) value() int64 {
 	// Increase the value for the number of dependents, so that
 	// tasks (and units) which block other tasks run before tasks
 	// that don't block other tasks.
-	value += priority * (u.NumDependents / length)
+	value += priority * (u.Settings.GetNumDependentsFactor() / 10) * (u.NumDependents / length)
```
Comment (ablack12) [inline scheduler/planner.go:261]: I feel like this shouldn't also consider priority, since we consider it above. Perhaps just multiplying num dependents by factor or something might be more straightforward (unless you have an idea why we consider overall length here)

File: model/distro/distro.go (hunk 1 above)
Comment (ablack12) [inline model/distro/distro.go:352]: Preference for this to still default to 1, even if that's different than what we have now. we could run an update to set this to 10 in the DB so that it's ready when we deploy.

File: service/templates/admin.html [hunk 3]
```diff
@@ -805,6 +805,11 @@ <h2 class="modal-title">[[modalTitle]]</h2>
 										<input type="number" step="1" min="0" max="100"
 												 ng-model="Settings.scheduler.stepback_task_factor">
 									</md-input-container>
+									<md-input-container class="control" style="width:45%;">
+										<label>Num Dependents Factor (0 to 100 inclusive)</label>
```
Comment (ZackarySantana) [inline service/templates/admin.html:809]: The code validation says 1 to 100. If we were to set it to 0, it would disable priority I think (because it would * 0). Not sure if we want that or this might be a typo

File: scheduler/planner.go (hunk 2 above)
Comment (Kimchelly) [inline scheduler/planner.go:261]: We should also probably add a test since there's some already in `scheduler/planner_test.go` covering rank value sorting behavior.

File: service/templates/admin.html (hunk 3 above)
Comment (ZackarySantana) [reply service/templates/admin.html:809]: Also applies to the min value of the input associated with the label below

File: scheduler/planner.go (hunk 2 above)
Comment (Kimchelly) [inline scheduler/planner.go:261]: If we want it to be multiplicative factor, could we just make it a float instead of an int? That way we can reduce the factor as much as we wanted (i.e. `0.1` would be equivalent).

**Review by Kimchelly — APPROVED** [review]
I made small suggestions but I think the change is fine as-is, especially if we expect this might come up again soon. Is there a ticket to eventually set this in the distro UI?

File: scheduler/planner.go (hunk 2 above)
Comment (Kimchelly) [reply scheduler/planner.go:261]: This is actually consistent with how priority behaves as a multiplicative factor. I don't think we would want this to be a special case for priority since the other factors account for priority.

Comment (hadjri) [reply scheduler/planner.go:261]: While I agree this function is hard to reason about, I think we should keep this line as is because it contains this PR to only changing the scalar effect of how many dependents a task has, rather than changing the logical formula of how the rank value is calculated.

**Comment by hadjri** [comment]
> I made small suggestions but I think the change is fine as-is, especially if we expect this might come up again soon. Is there a ticket to eventually set this in the distro UI?

I filed [DEVPROD-12095](https://jira.mongodb.org/browse/DEVPROD-12095)

File: cloud/parameterstore/fakeparameter/fake_parameter.go [hunk 4]
```diff
@@ -3,12 +3,12 @@ package fakeparameter
 import (
 	"context"
 	"flag"
+	"github.com/mongodb/grip"
```
Comment (ablack12) [inline cloud/parameterstore/fakeparameter/fake_parameter.go:6]: should move back

File: validator/distro_validator.go [hunk 5]
```diff
@@ -467,6 +467,12 @@ func ensureHasValidPlannerSettings(ctx context.Context, d *distro.Distro, s *eve
 			Level:   Error,
 		})
 	}
+	if settings.NumDependentsFactor < 0 || settings.NumDependentsFactor > 100 {
+		errs = append(errs, ValidationError{
+			Message: fmt.Sprintf("invalid planner_settings.num_dependents_factor value of %v for distro '%s' - its value must be a non-negative integer between 0 and 100, inclusive", settings.NumDependentsFactor, d.Id),
```
Comment (ablack12) [inline validator/distro_validator.go:472]: this can be a float

File: service/templates/admin.html (hunk 3 above)
Comment (ablack12) [reply service/templates/admin.html:809]: Can we say that this is a float here?

**Review by ablack12 — APPROVED** [review]
